In [1]:
import sys
import os
sys.path.append(os.path.abspath('..'))

from sklearn.datasets import load_iris
from src.models import get_logistic_model, get_knn_model
from src.trainer import train_model, model_evaluation, prediction

### Load Dataset

In [2]:
iris = load_iris()
X = iris.data
y = iris.target

### Load and Train Models
Will be using LogisticRegression and KNeighborsClassifier

In [3]:
# Logistic Regression
Logistic = get_logistic_model()
Logistic, X_train, X_test, y_train, y_test = train_model(Logistic, X, y)

# KNN
KNN = get_knn_model()
KNN, X_train, X_test, y_train, y_test = train_model(KNN, X, y)

### KNN Evaluation

In [4]:
acc, cm, cr = model_evaluation(KNN,X_test,y_test)
print(cr)

              precision    recall  f1-score   support

           0       1.00      1.00      1.00        10
           1       1.00      1.00      1.00        10
           2       1.00      1.00      1.00        10

    accuracy                           1.00        30
   macro avg       1.00      1.00      1.00        30
weighted avg       1.00      1.00      1.00        30



### Logistic Evaluation

In [5]:
acc, cm, cr = model_evaluation(Logistic, X_test, y_test)
print(cr)

              precision    recall  f1-score   support

           0       1.00      1.00      1.00        10
           1       1.00      0.90      0.95        10
           2       0.91      1.00      0.95        10

    accuracy                           0.97        30
   macro avg       0.97      0.97      0.97        30
weighted avg       0.97      0.97      0.97        30



### Comparison
While both models have great performance, KNN has a perfect score compared to Logistic.

KNN = 100%, Logistic = 97%

This is most likely due to stratify=y, because when testing without the parameter, both have gotten 100%